### ISSUE TITLE
[Question] Specific interface for Config Context

### ISSUE THREAD

**Issue by Domisiding** [TheNetworkGuy/netbox-zabbix-sync#40]
Hello,
I wanted to ask if you can also select an interface for the individual interfaces with this script in the config context.

Example:

Zabbix Agent on Interface 01
SNMP on interface 02

**Comment by TheNetworkGuy** [comment]
I think that this is easier then expacted since the interface functionality class is decoupled from the device class. However i'm unavailable for next weekend. Please give me some time to think about it and i'll come back to this suggestion asap :)

**Comment by Domisiding** [comment]
Hello,
I wanted to ask what the status is and whether a possibility has been found.

**Comment by TheNetworkGuy** [comment]
To follow up on this request: There is a huge difference on whether the individual interfaces should also have unique IP addresses. Is this also something that you want in your use case? Or does the functionallity implemented in https://github.com/netbox-community/netbox/issues/8137 cover that use case?

Furthermore could you give an example of your use case? For example a server with 1 management address and one for IPMI?

**Comment by Domisiding** [comment]
Hello,
For example, if you have a server that is provisioned via "iDRAC". You have one IP for the "iDRAC" and one IP for the regular server.

For example, if you want to monitor the iDRAC and the things on the server. If you have the iDRAC on interface 1, SNMP should be running there and the main (primary) IP of the Zabbix agent on interface 2.

192.0.0.1 (Interface 1)
192.0.0.2 (Interface 2 | Primary)

**Comment by TheNetworkGuy** [comment]
Okay so with the new out-of-band management IP address field this is a lot simpeler. However i'm struggeling with the logic of interface adjustments. When an IP address changes, which one of the 2 interfaces should be adjusted? Or if the protocol switches from agent to SNMP, how would the sync script have any idea which interface to adjust? Stretching this issue further templates are linked to interfaces for certain functons. So Zabbix could complain about resetting an interface (or both interfaces.) I'll look into this but unfortunately i can't promise much.

A workaround for now (although a bit cumbersome) would be to have 2 hosts for a server. One for the agent coming from Netbox and 1 manually configured for the IPMI.

**Comment by q1x** [comment]
I second the idea of using OOB IP as a secondary interface in Zabbix.

**Comment by q1x** [comment]
#122 might be of help here

**Comment by q1x** [comment]
This is ready in [develop](https://github.com/TheNetworkGuy/netbox-zabbix-sync/tree/develop).
Anyone willing to test: we look forward to your feedback.

**Comment by q1x** [comment]
@mathieumd @Domisiding do you see a chance to test our development code to see if it fits your needs?

**Comment by mathieumd** [comment]
@q1x Yes, I'm hoping to try it out tomorrow... if work and colleagues let me free time for it ;)

**Comment by mathieumd** [comment]
Can't make it work :-(

Had to:
- Run `.venv/bin/pip --require-virtualenv install -r requirements.txt` to install missing new deps.
- Add permissions `hostinterface.create` and `hostinterface.delete` to dedicated Zabbix User Role.

However, it won't create an IPMI interface, but an SNMP one. How am I supposed to tell it it should be an IPMI? I tried to define the `ipmi` dictionnary in NB Config Context (to set `username` and `pasword`, even though they'll be already set in our Zabbix template). Here is my Config Context:

```json
    "zabbix": {
        "interface_type": 1,
        "ipmi": {
            "password": "{$IPMI.PASSWORD}",
            "username": "{$IPMI.USER}"
        }
    }
```

The log:
```
2026-05-05 17:05:34,622 - NetBox-Zabbix-sync - INFO - Host pve-test01: Added interface with data {'main': '1', 'useip': '1', 'dns': '', 'ip': '10.x.y.86', 'type': '2', 'port': '161', 'details': {'version': '2', 'community': '{$SNMP_COMMUNITY}', 'bulk': '1'}, 'hostid': '10872'}.
```

**Comment by q1x** [comment]
Hi @mathieumd , if you have set the primary and oob ips on the device, this config context should create a primary agent interface and a secondary IPMI interface:

```json
{
    "zabbix": {
        "interface_type": "agent",
        "oob_interface_type": "ipmi",
        "ipmi": {
            "password": "{$IPMI.PASSWORD}",
            "username": "{$IPMI.USER}"
        }
    }
}
```

You'll need to set  `oob_sync`  to `True` in your config file as well.

**Comment by mathieumd** [comment]
Thanks, it's working after adding `oob_interface_type`!

However, I stumbled on a question: are Templates synchronized *before* Interfaces?

My Config Context is:
```json
{
    "zabbix": {
        "interface_type": "agent",
        "ipmi": {
            "password": "{$IPMI.PASSWORD}",
            "username": "{$IPMI.USER}"
        },
        "oob_interface_type": "ipmi",
        "templates": [
            "Custom Linux",
            "Custom IPMI"
        ]
    }
}
```

When I run it, it yields:

```
2026-05-07 15:34:50,191 - NetBox-Zabbix-sync - INFO - Host pve-test01: Template(s) OUT of sync.
2026-05-07 15:34:50,221 - NetBox-Zabbix-sync - ERROR - Host pve-test01: Unable to update. Zabbix returned the following error: Invalid params. Cannot inherit item with key "ipmi.get" of template "Custom IPMI" to host "pve-test01", because a host interface of type "IPMI" is required..
```

Which makes sense, because the IPMI interface has not been added.

Buf if I remove the "Custom IPMI" template from the Config Context, then it's adding the IPMI interface, and only then can I add the template back to the Config Context.

It feels like Interfaces should be sync'ed before Templates, no?

**Comment by q1x** [comment]
Thanks for testing :)
 
> It feels like Interfaces should be sync'ed before Templates, no?

Good point! Fixed via #209. Can you retry with the latest develop?

**Comment by mathieumd** [comment]
Quite better! ;-)

I only tested on a single machine, I hope to apply to all of them next week.

Thank you.

**Comment by mathieumd** [comment]
I wonder if it would not be better to move `oob_sync` to NetBox-side instead of `config.py`?

I have a physical host only monitored by ICMP, so it's default Zabbix interface type is SNMP. Enabling `oob_sync` triggers:
```
2026-05-11 11:17:37,259 - NetBox-Zabbix-sync - ERROR - Duplicate interface types found.
2026-05-11 11:17:37,259 - NetBox-Zabbix-sync - ERROR - Inconsistent interface configuration for host example01.
```
It's Config Context is as simple as this:
```json
    "zabbix": {
        "templates": [
            "Ping"
        ]
    }
```

**Comment by q1x** [comment]
You'll need to make sure that the Primary IP interface and the OOB IP interface are set to different types as we currently do not support multiple interfaces of the same type for syncing.

Can you describe the configuration for this device in NetBox and it's rendered config_context so I can understand this specific case a little better?

**Comment by mathieumd** [comment]
In NetBox, as a basis, we only define a default template (`Ping`) which applies to all hosts (we have other Config Contexts with higher priority for Linux Agent defining `interface_type="agent"`, for example). Therefore, the first Zabbix interface defaults to the default value of SNMP; and because `oob_sync` is enabled for all hosts actually having an OOB IP (which is the case for this host), but don't have any `oob_interface_type` defined (because it does not have any Config Contexts targeting it -- too bad CC cannot target devices with or without OOB IP), it gets a second Zabbix interface which defaults too to SNMP. Which leads to this `Duplicate interface types found`.

I wondered if moving `oob_sync` to Config Context would help to be more precise in such cases.

**Comment by q1x** [comment]
Thanks, I better understand the issue. I wasn't expecting OOB interfaces that wouldn't be need to monitored. Let me try and find a solution :)

**Comment by q1x** [comment]
@mathieumd sorry for the delay! 
I've been thinking over this issue and I think the cleanest solution would be to add logic to allow for override of the `interface_type` and `oob_interface_type` by the value `none` to disable that interface for use in Zabbix, like this:

```json
{
    "zabbix": {
        "interface_type": "agent",
        "oob_interface_type": "none"
   }
}
```

You could then use config context rules to disable OOB interfaces on hosts that do have OOB IP set but don't need to be monitored (based on device type, role or platform for instance).
Would that work for you?

**Comment by mathieumd** [comment]
Looks good to me.
In my case, I would set `"oob_interface_type": "none"` by default and override it for hosts I want to monitor.
